File: Api_ParkingHole/src/main/java/com/parkinghole/api_parkinghole/modelos/Coche.java
package com.parkinghole.api_parkinghole.modelos;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "coche")
@Data
public class Coche {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long cid;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "uid_dueno", nullable = false)
    @com.fasterxml.jackson.annotation.JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
    private Usuario dueno;

    private String marca;
    private String modelo;

    @Column(unique = true)
    private String matricula;

    private String color;
    private String imagenUrl;
    private boolean activo = true;

    private String capacidad;
}

File: TFG_app/app/src/main/java/com/example/tfg/models/Coche.kt
package com.example.tfg.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Coche(
    @SerializedName("cid")
    val cid: Long? = null,

    @SerializedName("dueno")
    val dueno: Usuario? = null,

    @SerializedName("marca")
    val marca: String? = "Desconocida",

    @SerializedName("modelo")
    val modelo: String? = "Desconocido",

    @SerializedName("matricula")
    val matricula: String? = "Sin matrícula",

    @SerializedName("color")
    val color: String? = "No especificado",

    @SerializedName("imagenUrl")
    val imagenUrl: String? = null,

    @SerializedName("activo")
    val activo: Boolean? = true,

    @SerializedName("capacidad")
    val capacidad: String? = null
) : Parcelable